dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/BitBucketSourceCodeRepo.java

```diff
@@ -219,44 +219,28 @@ public void updateReferenceType(String repositoryId, Version version) {
         if (version.getReferenceType() != Version.ReferenceType.UNSET) {
             return;
         }
+        String workspace = repositoryId.split("/")[0];
+        String repoSlug = repositoryId.split("/")[1];
+        String name = version.getReference();
         RefsApi refsApi = new RefsApi(apiClient);
         try {
-            PaginatedBranches paginatedBranches = refsApi
-                .repositoriesUsernameRepoSlugRefsBranchesGet(repositoryId.split("/")[0], repositoryId.split("/")[1]);
-            while (paginatedBranches != null) {
-                if (paginatedBranches.getValues().stream().anyMatch(key -> key.getName().equals(version.getReference()))) {
-                    version.setReferenceType(Version.ReferenceType.BRANCH);
-                }
-                if (paginatedBranches.getNext() != null) {
-                    paginatedBranches = getArbitraryURL(paginatedBranches.getNext(), new GenericType<PaginatedBranches>() {
-                    });
-                } else {
-                    paginatedBranches = null;
-                }
-            }
+            refsApi.repositoriesUsernameRepoSlugRefsBranchesNameGet(workspace, version.getReference(), repoSlug);
+            version.setReferenceType(Version.ReferenceType.BRANCH);
+            return;
         } catch (ApiException e) {
             LOG.error(gitUsername + ": apiexception on reading branches" + e.getMessage(), e);
             // this is not so critical to warrant a http error code
         }
 
         try {
-            PaginatedTags paginatedTags = refsApi
-                .repositoriesUsernameRepoSlugRefsTagsGet(repositoryId.split("/")[0], repositoryId.split("/")[1]);
-            while (paginatedTags != null) {
-                if (paginatedTags.getValues().stream().anyMatch(key -> key.getName().equals(version.getReference()))) {
-                    version.setReferenceType(Version.ReferenceType.TAG);
-                }
-                if (paginatedTags.getNext() != null) {
-                    paginatedTags = getArbitraryURL(paginatedTags.getNext(), new GenericType<PaginatedTags>() {
-                    });
-                } else {
-                    paginatedTags = null;
-                }
-            }
+            refsApi.repositoriesUsernameRepoSlugRefsTagsNameGet(workspace, name, repoSlug);
+            version.setReferenceType(Version.ReferenceType.TAG);
+            return;
         } catch (ApiException e) {
             LOG.error(gitUsername + ": apiexception on reading tags" + e.getMessage(), e);
             // this is not so critical to warrant a http error code
         }
+        throw new CustomWebApplicationException("Not a Bitbucket branch or tag", HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
Could be a 4xx, although maybe only the caller will know for sure.

Reply:
This function is called in a refresh endpoint. It's likely our problem instead of the user.